gson/src/main/java/com/google/gson/JsonPrimitive.java

```diff
@@ -95,9 +95,10 @@ public boolean isBoolean() {
   }
 
   /**
-   * convenience method to get this element as a boolean value.
-   *
-   * @return get this element as a primitive boolean value.
+   * Convenience method to get this element as a boolean value.
+   * If this primitive {@linkplain #isBoolean() is not a boolean}, the string value
+   * is parsed using {@link Boolean#parseBoolean(String)}. This means {@code "true"} (ignoring
+   * case) is considered {@code true} and any other value is considered {@code false}.
```

Comment:
Well that's pretty terrible. But I suppose we're stuck with it.

Reply:
Yeah..., I had created #1775 for that in the past (which also covers the type adapter for `Boolean`). Maybe we can revisit this behavior at some point. Do you think I should omit this detailed documentation here to make it easier to change this behavior in the future? Currently the behavior is simply undocumented.

Reply:
No, the current behaviour is unfortunate, but I think changing it would be too incompatible. So documenting it looks like the right thing to do.